exploration/chirps_gefs.md

```diff
@@ -0,0 +1,180 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.16.1
+  kernelspec:
+    display_name: ds-aa-cub-hurricanes
+    language: python
+    name: ds-aa-cub-hurricanes
+---
+
+# CHIRPS-GEFS
+
+Process CHIRPS-GEFS
+
+```python
+%load_ext jupyter_black
+%load_ext autoreload
+%autoreload 2
+```
+
+```python
+import pandas as pd
+import xarray as xr
+import ocha_stratus as stratus
+from sqlalchemy import text
+from tqdm.auto import tqdm
+from azure.core.exceptions import ResourceNotFoundError
+
+from src.datasources.chirps_gefs import ChirpsGefsConfig, ChirpsGefsLoader
+from src.datasources import codab
+from src.constants import *
+from src.utils.database import create_chirps_gefs_table
+```
+
+```python
+TABLE_NAME = f'{PROJECT_PREFIX.replace("-", "_")}_chirps_gefs'
+```
+
+```python
+TABLE_NAME
+```
+
+```python
+# create projects schema - this should never need running again
+# try:
+#     with stratus.get_engine("dev", write=True).begin() as conn:
+#         conn.execute(text(f"CREATE SCHEMA IF NOT EXISTS {SCHEMA_NAME}"))
+# except Exception as e:
+#     print("Schema creation failed:", e)
+```
+
+```python
+# create table in projects schema
+# create_chirps_gefs_table(TABLE_NAME, stratus.get_engine("dev", write=True))
+```
+
+```python
+adm0 = codab.load_codab_from_blob()
+```
+
+```python
+adm0.plot()
+```
+
+```python
+config = ChirpsGefsConfig(geometry=adm0, region_name="databricks_run")
+loader = ChirpsGefsLoader(config)
+```
+
+```python
+# check which dates we already have processed
+query = f"""
+SELECT issued_date, COUNT(*) AS n_entries
+FROM projects.{TABLE_NAME}
+GROUP BY issued_date
+ORDER BY issued_date;
+"""
+
+df_existing_dates = pd.read_sql(
+    query, stratus.get_engine("dev"), parse_dates="issued_date"
+)
+```
+
+```python
+df_existing_dates
+```
+
+```python
+# check complete dates - we have to do this because some dates might not have all the stats
+df_complete_dates = df_existing_dates[
+    df_existing_dates["n_entries"] == df_existing_dates["n_entries"].max()
+]
```

Comment:
i might be missing something, but the order of operations is a little confusing. It seems like the commented out code above is creating an empty table, then the for-loop below (line 136) is populating the table. But this code above is supposed to be reading in the table from the already populated data. So i think this should probably be moved dowb?

Reply:
if i'm right about that I'd probably do something like create a script 

```
#> data-raw/01_create_chirps_gefs_zonal_postgres
create_chirps_gefs_table(TABLE_NAME, stratus.get_engine("dev", write=True))

#> forloop code
```

Than in this exploration notebook, i'd just have the code to check the dates etc?

Reply:
ah wait i think i understand that the order is actually fine, as the first time it's run it will pick up the empty table and see that all the dates are missing. As this notebook is actually more like a prod data downloading script/notebook i think it makes sense to move it somewhere else. You could just put a condition on create_chirps_gefs_table to not allow it to be created if already in DB, as well as wrap the schema creation process in a similar function. That way they could sit at top of the code without being commented out. Maybe for python projects, we've typically been putting these in `analysis` ?